### PR TITLE
Adds the Airlock Authentication Override Card

### PR DESCRIPTION
**Pull request by MrDoomBringer**
## About The Pull Request
Cryptographic sequencers can no longer open airlocks
The Airlock Authentication Override Card can open airlocks (3 charges, each charge refills over the course of a couple minutes)

## Why It's Good For The Game
@optimumtact said it was good for the game
The data says the emag is the most bought traitor item. This is because the emag has too much bloat. To help mitigate this this we've taken away airlock hacking from the emag, however instead of removing the feature outright (as in #50632) we move the feature to a new item, for those who really still want to be able to hack doors

## Changelog
:cl: MrDoomBringer with some code stolen from itseasytosee
add: added the Airlock Authentication Override Card. Use it to open airlocks!
add: added Syndicate Jaws of Life to the nuke op uplink. Use it in it's crowbar configuration to open as many doors as you'd like!
del: The cryptographic sequencer can no longer open airlocks :(
/:cl:

Closes #50632

### CODE REVIEW

**Comment by EgorDinamit** [comment]
Please don't. 
**Also put a valid reason for why it is good for the game.**

File: code/game/objects/items/emags.dm [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/* Emags
```
Comment (Fikou) [inline code/game/objects/items/emags.dm:1]: you forgot the pumpkin emag here

**Comment by Megarop** [comment]
I can get not wanting traitors to be able to open as many doors as they like with an emag but this is a pretty harsh nerf for nuke ops

Also the Airlock Authentication Override card sounds awful, especially for a 6 TC item that you have to dump even more TC in in order to use it more than _five_ times

**Comment by MrDoomBringer** [comment]
@EgorDinamit ok
@Megarop sorry, didn't change TC values. It'll be updated in a second

**Comment by TheVekter** [comment]
Yeah, I'm not sure how I feel about this. oranges can put their two cents in on this, but I'm not a fan. If we're gonna nerf the emag, we probably need a better solution than "an emag but worse".

**Comment by Shadowflame909** [comment]
why would I spend 3 tc, (plus more if I want more charges.) on a limited use door opener, when I could spend 8 tc on an e-sword, that's a lethal weapon which doubles as a door opener.

The reason the e-mag was so good, was because it was the best door-opener, and also doubled as a non-lethal sabotage option.

There are so many more lethal options, cheaper depending on what you're trying to do. Like buying x4, or getting the agent card which allows you to kill people and steal their access.

I think this is useless and at the very worse, a gross waste of a new player's tc. 

If someone really wants 5 magic door opening chances. They'll just buy a bag of c4. If they want to spend more tc on more charges, they'll buy the c4 individually.

But what I think the majority of experienced players are going to do? Much more cost-expensive tc wise to break down a door with a very strong weapon, like the infinite use dual e-sword or a holoparasite, then waste tc on this.

**Comment by wesoda25** [comment]
I think you should just consolidate this into the emag so that it has unlimited uses on machines, but with the limited uses for doors. Also one tc per door is way too steep a price, it should be double or triple. Still I don’t like this, there is no need for this PR since there was no guarantee the other would be merged.

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1359,7 +1359,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 			in electronic devices, subverts intended functions, and easily breaks security mechanisms."
 	item = /obj/item/card/emag
-	cost = 6
+	cost = 4
+
+/datum/uplink_item/device_tools/doormag
+	name = "Airlock Authentication Override Card"
+	desc = "It's a specialized cryptographic sequencer specifically designed to override station airlock access codes. \
+			Seems to crack airlock encryption algorithms by using raw telecrystals."
+	item = /obj/item/card/emag
```
Comment (Qustinnus) [inline code/modules/uplink/uplink_items.dm:1368]: wrong type chumbis

**Comment by Megarop** [comment]
> The data says the emag is the most bought traitor item. This is because the emag has too much bloat.

Why don't you remove the 'bloat' then instead of removnig the main function of the emag so it has nothing _but_ 'bloat'?

> @Megarop I made the PR without tweaking the TC values first. Take a look now, prices are more reasonable

I'm still against the entire concept of having a traitor item that you need to constantly insert TC into to keep using like it's an arcade machine but I guess this is slightly more reasonable

> one tc per door is way too steep a price, it should be double or triple.

I just assumed that was a placeholder and they were gonna change it later, with that price it's cheaper to just buy another card over and over again than to put TC in the one you already have.

**Comment by necromanceranne** [comment]
I actually hate this more than outright nothing. Generally speaking, emags were a convenience item at best and could be replaced with a RCD (kinda). Meanwhile, for ops, it is somewhat mandatory but also enables blitz ops off the back of the item alone. For ops, this basically means nothing and they are the biggest offenders for using emags for quick mobility. 

If anything this is a buff to blitz ops since about 5 doors is all you'll need to hit to reach the captain with his pants down.

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (MrDoomBringer) [reply code/modules/uplink/uplink_items.dm]: grr bark bark

**Comment by ATH1909** [comment]
If you really HAVE to do this, why not make this work like fulpstation's budget emag, in that it has only two charges, but regenerates one of its charges every 30 seconds?

**Comment by MrDoomBringer** [comment]
@ATH1909 I've gotten a couple suggestions about making it a time-based recharge with a certain number of charges, I'll look into that

**Comment by ATH1909** [comment]
For reference, here's how fulp did it:
https://github.com/TheSwain/Fulpstation/pull/273
https://github.com/TheSwain/Fulpstation/pull/295

**Comment by AdamElTablawy** [comment]
If i'm being made to spend 6 tc on an item that has no use other than opening doors, I'm buying a 1tc syndie toolbox and going from there.

**Comment by optimumtact** [comment]
that is pretty much what the goal is mate

**Comment by optimumtact** [comment]
I'd prefer they use the holoparasite, the emag is stale.

**Comment by MrDoomBringer** [comment]
@Shadowflame909 Spending 18 tc on something that will tell everyone you're a traitor that can do nothing put punch is not really an option when you want to stealthily break into a department, you know

**Comment by BeeSting12** [comment]
Keep this rolled into the emag but make it rechargeable like the EMP flash light. Four doors and then you have to wait for it to recharge.

**Comment by optimumtact** [comment]
Discuss it here instead
https://tgstation13.org/phpBB/viewtopic.php?f=5&t=26520&p=557923#p557923

**Comment by saprasam** [comment]
revert this change right now

**Comment by optimumtact** [comment]
No!

**Comment by saprasam** [comment]
if you dont i'm gonna call you stupid

**Comment by MrDoomBringer** [comment]
No! You're stupid!

**Comment by saprasam** [comment]
no, you're incomprehensibly dumb. i've seen thumbtacks smarter then you, if that dumb PINK HAIR got shaved off the size of your head would be similar to my TOE.

**Comment by ujiohiugyihljhujh** [comment]
you all are stupid

**Comment by MrDoomBringer** [comment]
image
detected

**Pull request merged**